Index: ../common/TLVBuffer.h
===================================================================
--- ../common/TLVBuffer.h	(revision 711)
+++ ../common/TLVBuffer.h	(working copy)
@@ -33,6 +33,8 @@
 #include <map>
 #include "TLV.h"
 
+__asm__(".symver memcpy,memcpy@GLIBC_2.2.5");
+
 namespace eIDMW
 {
 
Index: ../common/ByteArray.h
===================================================================
--- ../common/ByteArray.h	(revision 711)
+++ ../common/ByteArray.h	(working copy)
@@ -23,6 +23,8 @@
 
 #include <string>
 
+__asm__(".symver memcpy,memcpy@GLIBC_2.2.5");
+
 namespace eIDMW
 {
 
Index: ../common/libtomcrypt/tomcrypt_hash.h
===================================================================
--- ../common/libtomcrypt/tomcrypt_hash.h	(revision 711)
+++ ../common/libtomcrypt/tomcrypt_hash.h	(working copy)
@@ -26,6 +26,8 @@
 extern "C" {
 #endif
 
+__asm__(".symver memcpy,memcpy@GLIBC_2.2.5");
+
 /* ---- HASH FUNCTIONS ---- */
 #ifdef USE_SHA512
 struct sha512_state {
Index: ../common/libtomcrypt/tomcrypt_macros.h
===================================================================
--- ../common/libtomcrypt/tomcrypt_macros.h	(revision 711)
+++ ../common/libtomcrypt/tomcrypt_macros.h	(working copy)
@@ -206,6 +206,8 @@
 
 #elif defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__)) && !defined(INTEL_CC) && !defined(LTC_NO_ASM)
 
+__asm__(".symver memcpy,memcpy@GLIBC_2.2.5");
+
 static inline unsigned ROL(unsigned word, int i)
 {
    asm ("roll %%cl,%0"
