Index: ../cardlayer/PkiCard.cpp
===================================================================
--- ../cardlayer/PkiCard.cpp	(revision 2097)
+++ ../cardlayer/PkiCard.cpp	(working copy)
@@ -443,7 +443,7 @@
 
 	MWLOG(LEV_INFO, MOD_CAL, L"     No SSO: ask PIN and sign (key: ID=0x%0x, algo=0x%0x, "
 			L"%d bytes input)", key.ulID, algo, oData.Size());
-	return SignInternal(key, algo, oData, &Pin);
+	return SignInternal(key, algo, oData);
 
 }
 
