diff --git a/pteid-mw-pt/_src/eidmw/common/ConfigCommon.cpp b/pteid-mw-pt/_src/eidmw/common/ConfigCommon.cpp
--- a/common/ConfigCommon.cpp
+++ b/common/ConfigCommon.cpp
index eb21c51c4..e99b6493f 100644
@@ -53,7 +53,7 @@ const struct CConfig::Param_Str CConfig::EIDMW_CONFIG_PARAM_GENERAL_CERTS_DIR =
 #else
 const struct CConfig::Param_Str CConfig::EIDMW_CONFIG_PARAM_GENERAL_CERTS_DIR = {
 	EIDMW_CNF_SECTION_GENERAL, EIDMW_CNF_GENERAL_CERTSDIR,
-	WDIRSEP L"usr" WDIRSEP L"local" WDIRSEP L"share" WDIRSEP L"certs" WDIRSEP};
+	WDIRSEP L"app" WDIRSEP L"share" WDIRSEP L"certs" WDIRSEP};
 #endif
 #ifdef WIN32
 const struct CConfig::Param_Str CConfig::EIDMW_CONFIG_PARAM_GENERAL_CERTS_DIR_TEST = {
@@ -61,7 +61,7 @@ const struct CConfig::Param_Str CConfig::EIDMW_CONFIG_PARAM_GENERAL_CERTS_DIR_TE
 #else
 const struct CConfig::Param_Str CConfig::EIDMW_CONFIG_PARAM_GENERAL_CERTS_DIR_TEST = {
 	EIDMW_CNF_SECTION_GENERAL, EIDMW_CNF_GENERAL_CERTSDIR_TEST,
-	WDIRSEP L"usr" WDIRSEP L"local" WDIRSEP L"share" WDIRSEP L"certs_test" WDIRSEP};
+	WDIRSEP L"app" WDIRSEP L"share" WDIRSEP L"certs_test" WDIRSEP};
 #endif
 #ifdef WIN32
 const struct CConfig::Param_Str CConfig::EIDMW_CONFIG_PARAM_GENERAL_WEB_DIR = {
@@ -69,7 +69,7 @@ const struct CConfig::Param_Str CConfig::EIDMW_CONFIG_PARAM_GENERAL_WEB_DIR = {
 #else
 const struct CConfig::Param_Str CConfig::EIDMW_CONFIG_PARAM_GENERAL_WEB_DIR = {
 	EIDMW_CNF_SECTION_GENERAL, EIDMW_CNF_GENERAL_WEBDIR,
-	WDIRSEP L"usr" WDIRSEP L"local" WDIRSEP L"share" WDIRSEP L"pteid-mw" WDIRSEP L"www"};
+	WDIRSEP L"app" WDIRSEP L"share" WDIRSEP L"pteid-mw" WDIRSEP L"www"};
 #endif
 
 const struct CConfig::Param_Str CConfig::EIDMW_CONFIG_PARAM_GENERAL_LANGUAGE = {EIDMW_CNF_SECTION_GENERAL,
diff --git a/pteid-mw-pt/_src/eidmw/pteid-mw.pro b/pteid-mw-pt/_src/eidmw/pteid-mw.pro
--- a/pteid-mw.pro
+++ b/pteid-mw.pro
index 24b96f264..31284fd5b 100644
@@ -35,10 +35,10 @@ CONFIG += ordered
 
 QMAKE_STRIP = echo
 
-data.path +=  /usr/local/share/certs
+data.path += /app/share/certs
 data.files += misc/certs/*
 
-web.path += /usr/local/share/pteid-mw/www
+web.path += /app/share/pteid-mw/www
 web.files += misc/web/*
 
 INSTALLS += data \
