diff --git a/pteid-mw-pt/_src/eidmw/pteid-mw.pro b/pteid-mw-pt/_src/eidmw/pteid-mw.pro
index 24b96f264..f334983b7 100644
--- a/pteid-mw.pro
+++ b/pteid-mw.pro
@@ -20,8 +20,7 @@ SUBDIRS += cardlayer/cardlayer.pro
 SUBDIRS += pkcs11/pkcs11.pro \
            applayer \
            CMD/services/cmdServices.pro \
-           eidlib \
-           eidlibJava_Wrapper
+           eidlib
 
 applayer.depends = pteid-poppler		
 
