Index: ../cardlayer/cardpluginBeid/BeidCard.cpp
===================================================================
--- ../cardlayer/cardpluginBeid/BeidCard.cpp	(revision 711)
+++ ../cardlayer/cardpluginBeid/BeidCard.cpp	(working copy)
@@ -727,13 +727,13 @@
 
     // Pretty unique for smart cards: first MSE SET, then verify PIN
     // (needed for the nonrep key/pin, but also usable for the auth key/pin)
-    if (pPin != NULL)
-    {
-        unsigned long ulRemaining = 0;
-        bool bOK = PinCmd(PIN_OP_VERIFY, *pPin, "", "", ulRemaining, &key);
-        if (!bOK)
-			throw CMWEXCEPTION(ulRemaining == 0 ? EIDMW_ERR_PIN_BLOCKED : EIDMW_ERR_PIN_BAD);
-    }
+    // if (pPin != NULL)
+    // {
+    //     unsigned long ulRemaining = 0;
+    //     bool bOK = PinCmd(PIN_OP_VERIFY, *pPin, "", "", ulRemaining, &key);
+    //     if (!bOK)
+    // 			throw CMWEXCEPTION(ulRemaining == 0 ? EIDMW_ERR_PIN_BLOCKED : EIDMW_ERR_PIN_BAD);
+    // }
 
     SetSecurityEnv(key, algo, oData.Size());
 
